tern/analyze/docker/dockerfile.py

```diff
@@ -118,6 +120,31 @@ def parse_from_image(dfobj):
     return image_list
 
 
+def expand_from_images(dfobj):
+    '''Replace all parent_image values with their full tag@sha:digest value.
+    Update the structure dictionary with the same information.
+    dfobj: the Dockerfile object created using get_dockerfile_obj'''
+    # update parent_images
+    images = parse_from_image(dfobj)
+    for i, img in enumerate(images):
+        # don't re-pull digest if already available
+        if not img['digest_type'] and not img['digest']:
+            if not img['tag']:
+                img['tag'] = 'latest'
+            dfobj.parent_images[i] = container.get_image_digest(
+                container.get_image(img['name'] + tag_separator +
+                                    img['tag']))
```

Comment:
So it looks like here `dfobj.parent_images` has the image digests all set. Why not use these values in the loop below? To maintain association, you can create a dictionary with the original name and the digest name.

Reply:
Also, what happens when `get_image` returns `None` here?

Reply:
I already made changes in the loop below to use the `dfobj.parent_images`.

I don't have a check to account for None image types... was thinking this would happen at a higher level but can add one here if you like.

Reply:
Sorry, it looks like it preserved my original review after you made the change. I would suggest checking if what is returned by `check_image` is not None and then doing the replacement and if not, log the error and don't expand.